node/src/protocol/cryptography.rs

```diff
@@ -118,6 +122,13 @@ impl CryptographicProtocol for GeneratingState {
                         public_key = hex::encode(r.public_key.to_bytes()),
                         "successfully completed key generation"
                     );
+                    ctx.secret_storage()
```

Comment:
So if this storage attempt fails, the progress() will fail even tho it actually completed key generation. Is there a problem with that?
Also want to check the purpose of this change? Is it to have the nodes generate keys and upload themselves instead of us giving them keys?

Reply:
Not ideal for sure, but I would say storage being unavailable is a fatal error. If we can't store the private key share, then we wouldn't be able to recover it after reboot.


> Also want to check the purpose of this change? Is it to have the nodes generate keys and upload themselves instead of us giving them keys?

So the idea is to be able to restart the node without loosing crucial information - current epoch, pre-agreed root public key and private key share specific for this node.

Reply:
`If we can't store the private key share, then we wouldn't be able to recover it after reboot.` -- Agree.
so when this progress() fails what happens next? Will progress() be tried again to generate and store the key share?

Reply:
Hmm, that's a good point. It might just make sense to panic here instead to signify that this is a fatal error. Otherwise progress will try to finish the protocol and save the result indefinitely.